File: Whois Client/cpp/whois.cpp
/*
 * IPWorks 2022 C++ Edition - Demo Application
 *
 * Copyright (c) 2023 /n software inc. - All rights reserved. - www.nsoftware.com
 *
 */


#include <stdio.h>
#include <string.h>
#include "../../include/ipworks.h"
#define LINE_LEN 100

int main(int argc, char *argv[])
{
	if (argc != 2) {

		fprintf(stderr, "usage: whois domain\n\n");
		fprintf(stderr, "  domain   the name of the domain to get information about\n");
		fprintf(stderr, "\nExample: whois google.com\n\n");
		printf("Press enter to continue.");
		getchar();

	}
	else{
		
		Whois whois;
		int ret_code = 0;

		whois.SetServer("whois.internic.net");
		ret_code = whois.Query(argv[1]);

		if (!ret_code)
		{

			printf(whois.GetDomainInfo());
			printf("This information provided by: %s.\n\n", whois.GetServer());

		}
		else
		{
			printf("error: %d (%s).\n", ret_code, whois.GetLastError());
		}

		printf("Press ENTER to continue...");
		getchar();
		printf("\n");
		return 0;
	}
	
}

File: Whois Client/java/src/whois.java
/*
 * IPWorks 2022 Java Edition- Demo Application
 *
 * Copyright (c) 2023 /n software inc. - All rights reserved. - www.nsoftware.com
 *
 */

import java.io.*;

import ipworks.*;


public class whois extends ConsoleDemo {
	
	public static void main(String[] args) {
		
		if(args.length!=1) {
			
			System.out.println("usage: whois domain");
			System.out.println("");
			System.out.println("  domain the name of the domain to get information about");			
			System.out.println("\r\nExample: whois google.com");
		
		} else {
			
			Whois whois1 = new Whois();
		    try {
		    	
		    	whois1.setServer("whois.internic.net");
		    	whois1.query(args[0]); // domain name

		    	System.out.println(whois1.getDomainInfo());
		    	System.out.println("This information provided by: " + whois1.getServer());
		    	
		    }catch (IPWorksException e) {
		    		displayError(e);
		    }			
		}    	  	  
	}
}

class ConsoleDemo {
  private static BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

  static String input() {
    try {
      return bf.readLine();
    } catch (IOException ioe) {
      return "";
    }
  }
  static char read() {
    return input().charAt(0);
  }

  static String prompt(String label) {
    return prompt(label, ":");
  }
  static String prompt(String label, String punctuation) {
    System.out.print(label + punctuation + " ");
    return input();
  }

  static String prompt(String label, String punctuation, String defaultVal)
  {
	System.out.print(label + " [" + defaultVal + "] " + punctuation + " ");
	String response = input();
	if(response.equals(""))
		return defaultVal;
	else
		return response;
  }

  static char ask(String label) {
    return ask(label, "?");
  }
  static char ask(String label, String punctuation) {
    return ask(label, punctuation, "(y/n)");
  }
  static char ask(String label, String punctuation, String answers) {
    System.out.print(label + punctuation + " " + answers + " ");
    return Character.toLowerCase(read());
  }

  static void displayError(Exception e) {
    System.out.print("Error");
    if (e instanceof IPWorksException) {
      System.out.print(" (" + ((IPWorksException) e).getCode() + ")");
    }
    System.out.println(": " + e.getMessage());
    e.printStackTrace();
  }
}




